### DISCUSSION TITLE
Access wagmi client from inside custom connector

### DISCUSSION THREAD

**Discussion by ghardin1314** [wevm/wagmi#1119] [Q&A]
I am writing my own custom wallet connector for the wagmi client. I see that many of the supported connectors (Metamask, WalletConnect) access the wagmi client directly using `getClient` that is an internal import. Is there a way to also include this functionality in my custom connector?

**Comment by egorbond31** [comment]
getClient in a connector has been using only for getting storage. Storage is a localStorage by default. So you can replace getClient().storage with window.localStorage . Note that you need to start the localStorage key with a prefix (default 'wagmi'). Your code will be look like:
const storage = window.localStorage; 
if you use SSR ->  const storage = typeof window != 'undefined' ? window.localStorage : noopStorage (noopStorage imported from '@wagmi/core')

storage?.setItem(`${storagePrefix}.${this.shimDisconnectKey}`, 'true');
storage?.getItem(`${storagePrefix}.${this.shimDisconnectKey}`)
